yara-x-cli/src/commands/fmt.rs

```diff
@@ -18,27 +19,38 @@ pub fn fmt() -> Command {
             arg!(-w  --write ... "Write output to source file instead of stdout")
                 .action(ArgAction::SetTrue),
         )
+        .arg(
+            arg!(-t  --test ... "Exit with failure if reformatting changed the file")
+                .action(ArgAction::SetTrue),
+        )
 }
 
 pub fn exec_fmt(args: &ArgMatches) -> anyhow::Result<()> {
     let files = args.get_many::<PathBuf>("FILE");
     let write = args.get_one::<bool>("write");
+    let test = args.get_one::<bool>("test");
 
     let formatter = Formatter::new();
 
     if let Some(files) = files {
         for file in files {
             let input = fs::read(file.as_path())?;
-            if *write.unwrap() {
-                let mut formatted = Cursor::new(Vec::new());
 
-                formatter.format(input.as_slice(), &mut formatted)?;
-                formatted.rewind()?;
+            let mut formatted = Cursor::new(Vec::new());
 
-                File::create(file.as_path())?
-                    .write_all(formatted.into_inner().as_slice())?;
+            formatter.format(input.as_slice(), &mut formatted)?;
+            formatted.rewind()?;
+
+            let output = formatted.into_inner();
+
+            if *test.unwrap() && input != output {
+                process::exit(3)
```

Comment:
Why not `process::exit(2)`? Exit code 2 is not used for any purpose yet. 

I'm also wondering if it wouldn't be better to return the path of the files that requires changes, as `gofmt -l` does. This provides more information to the user. If you call `yr fm` with multiple files, you may want to know which files are changed and which don't.

Reply:
Thank you for taking a look! I updated this to use exit code 2, and output a list of files that require changes. 

<img width="1391" alt="image" src="https://github.com/VirusTotal/yara-x/assets/21696437/0d804f89-51b5-4b27-8925-529f61e6ee23">